solutions/USACO dec18-bucketlist.mdx

```diff
@@ -41,3 +41,65 @@ file_out.close()
 </PySection>
 
 </LanguageSection>
+
+## Solution 2
+
+Similar to the second solution in the official editorial. 
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>  
+#include <cstdio>
+using namespace std;
+int event[1001]; // declaring the array globally to initialize all the elements to 0
+int main() {
+	freopen("blist.in", "r", stdin);
+	freopen("blist.out", "w", stdout); 
+	int maxtime = 1000;
+	int n; cin >> n;
+		
+		
+	for (int i = 0; i < n; i++) { // iterating through every cow
+		int start; cin >> start; 
+		int end; cin >> end;
+		int amt; cin >> amt;
+		// at time start, cow i needs to be milked, so 
+		// we'll need amt additional buckets (hence adding amt). 
+		// at time end, cow i is done milking, so we can free up 
+		// amt buckets (hence subtracting amt). This technique is called a 
+		// "sweep", because if you think of all the points
+		// on a horizontal number line, we are "sweeping" across 
+		// the line, processing each point that we come across.
```

Comment:
Is it worth keeping the "sweep" stuff in, or should we remove it? It's already described in the official editorial and it isn't necessary to understand the code, so I feel like it's sort of out of place here / it might confuse people.

@maggie-j-liu @dongliu0426

Reply:
yea i think "sweep" is kinda confusing for bronze

Reply:
^ i agree

if there's this many comments, they should probably be made into an editorial since it clutters the code

Reply:
gonna fix it

Reply:
> if there's this many comments, they should probably be made into an editorial since it clutters the code

^